indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillCategoryExtraction.java

```diff
@@ -0,0 +1,180 @@
+            }
+        }
+    }
+
+    /**
+     * Check if the confidence score satisfied the specified filter.
+     * @param confidence    Confidence score obtained for each category returned by the API
+     * @return              Returns true if the condition is satisfied, else returns false.
+     */
+    private boolean isFilterSatisfied(double confidence) {
+        if(confidence >= this.categoryConfidence)
+            return true;
+        else
+            return false;
+    }
+
+    /**
+     * Parses the Configuration for the Category Extraction Skill
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException
+     */
+    @Override
+    protected void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+        parseAISkillName((String) aiSkill.get(Constants.CONFIG_SKILL_NAME));
+        parseOutputMappings((JSONArray) aiSkill.get(Constants.CONFIG_OUTPUT_MAPPINGS), schema);
+        parseInputs((JSONObject) aiSkill.get(Constants.CONFIG_INPUTS));
+        parseFilter((JSONObject) aiSkill.get(Constants.CONFIG_FILTERS));
+    }
+
+    /**
+     * Execute the Category Extraction Skill and populates the structured Data.
+     * @param contentOrURI      The actual file content or Cloud storage URI
+     * @param structuredData    Multimap to store the structured data for indexing.
+     */
+    @Override
+    public void executeSkill(String contentOrURI, Multimap<String, Object> structuredData) {
+        try (LanguageServiceClient language = LanguageServiceClient.create()) {
+            Document doc = buildNLDocument(language, this.inputLanguage, contentOrURI);
+
+            ClassifyTextRequest request = ClassifyTextRequest.newBuilder().setDocument(doc).build();
+            ClassifyTextResponse response = language.classifyText(request);
+
+            for(OutputMapping outputMap : getOutputMappings()) {
+                String propertyName = outputMap.getPropertyName();
+                String fieldName = outputMap.getSkillOutputField();
+                switch(fieldName) {
+                    case Constants.CONFIG_CATEGORY: {
+                        for (ClassificationCategory category : response.getCategoriesList()) {
+                            if(isFilterSatisfied((double)category.getConfidence())) {
+                                structuredData.put(propertyName.split("\\.")[1],  category.getName());
+                            }
+                        }
+                        break;
+                    }
+                    default: {
+                        log.info("Output Field "+ fieldName + " not supported for AISkill Category Extraction. It will be ignored.");
+                    }
+                }
+            }
+            language.shutdown();
+            language.awaitTermination(30, TimeUnit.SECONDS);
+            language.close();
```

Comment:
maybe we should expose a function e.g. shutdown in driver which will internal shutdown all skills, each skill will shutdown the clients.

Reply:
I can add the shutdown() as part of the skill interface, that can take care of any clean up and call this from driver after executing the skill.